src/sentry/migrations/0760_add_org_to_api_auth.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 5.1.1 on 2024-09-13 18:10
+
+from django.db import migrations
+
+import sentry.db.models.fields.hybrid_cloud_foreign_key
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production.
+    # This should only be used for operations where it's safe to run the migration after your
+    # code has deployed. So this should not be used for most operations that alter the schema
+    # of a table.
+    # Here are some things that make sense to mark as post deployment:
+    # - Large data migrations. Typically we want these to be run manually so that they can be
+    #   monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   run this outside deployments so that we don't block them. Note that while adding an index
+    #   is a schema change, it's completely safe to run the operation after the code has deployed.
+    # Once deployed, run these manually via: https://develop.sentry.dev/database-migrations/#migration-deployment
+
+    is_post_deployment = False
+
+    dependencies = [
+        ("sentry", "0759_remove_spanattributeextraction_tables"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="apiauthorization",
+            name="organization_id",
+            field=sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                "sentry.Organization", db_index=True, null=True, on_delete="CASCADE"
+            ),
+        ),
+        migrations.AddField(
+            model_name="apigrant",
+            name="organization_id",
+            field=sentry.db.models.fields.hybrid_cloud_foreign_key.HybridCloudForeignKey(
+                "sentry.Organization", db_index=True, null=True, on_delete="CASCADE"
+            ),
+        ),
+        migrations.AddField(
+            model_name="apitoken",
+            name="scoping_organization_id",
```

Comment:
You'll also need a similar field on `ApiTokenReplica` so that authentication code will work in regions. Be careful though, the `ApiTokenReplica` already has an `organization` field on it, but that field is used for sentry app flows currently.

Reply:
ok that's very strange. Do you have more context? Why does a replica has a field that the original model doesn't have 😕 

I didn't call it `organization` specifically because ApiToken has a method `organization_id` that refers the organization from the other fields.

Reply:
If I'm understanding correctly, actually this will already set the `organization` on replica so I don't need to add it there: https://github.com/getsentry/sentry/blob/6bd307fc0a5263654d89605eca14343897468d78/src/sentry/auth/services/auth/serial.py#L82

Reply:
> ok that's very strange. Do you have more context? Why does a replica has a field that the original model doesn't have 😕

In order for `ApiTokenReplica` to also offer an `organization_id` attribute we synchronize the sentryapp derived organization state into the replica model so that we don't need to use RPC.

My previous comment was that we'll also need a `scoping_organization_id` on `ApiTokenReplica` so that endpoint access can be scoped in the region endpoints.

Reply:
In region endpoint when want to authenticate based on token, we look at `ApiTokenReplica.organization` and that would be the organization with ID `scoping_organization_id` because of this line: https://github.com/getsentry/sentry/pull/77491/files#r1771716626

I don't want to create another organization_id or scoping_organization_id field because if they become inconsistence we can't decide which one is the source of truth.